code/datums/skills.dm

```diff
@@ -1,529 +1,658 @@
 /datum/skills
-	var/name //the name of the skillset
-	var/cqc = SKILL_CQC_DEFAULT
-	var/melee_weapons = SKILL_MELEE_DEFAULT
-
-	var/firearms = SKILL_FIREARMS_DEFAULT
-	var/pistols = SKILL_PISTOLS_DEFAULT
-	var/shotguns = SKILL_SHOTGUNS_DEFAULT
-	var/rifles = SKILL_RIFLES_DEFAULT
-	var/smgs = SKILL_SMGS_DEFAULT
-	var/heavy_weapons = SKILL_HEAVY_WEAPONS_DEFAULT
-	var/smartgun = SKILL_SMART_DEFAULT
-	var/spec_weapons = SKILL_SPEC_DEFAULT
-
-	var/engineer = SKILL_ENGINEER_DEFAULT
-	var/construction = SKILL_CONSTRUCTION_DEFAULT
-	var/leadership = SKILL_LEAD_NOVICE
-	var/medical = SKILL_MEDICAL_UNTRAINED
-	var/surgery = SKILL_SURGERY_DEFAULT
-	var/pilot = SKILL_PILOT_DEFAULT
-	var/police = SKILL_POLICE_DEFAULT
-	var/powerloader = SKILL_POWERLOADER_DEFAULT
-	var/large_vehicle = SKILL_LARGE_VEHICLE_DEFAULT
-
-	var/list/values = list()
-
-
-/datum/skills/New()
-	values = list("Close Quarter Combat" = cqc, "Melee Weapons" = melee_weapons, "Firearms" = firearms, "Pistols" = pistols, "Shotguns" = shotguns,\
-	"Rifles" = rifles, "Submachine Guns" = smgs, "Heavy Weapons" = heavy_weapons, "Smartgun" = smartgun, "Specialist Weapons" = spec_weapons, \
-	"Engineering" = engineer, "Construction" = construction, "Leadership" = leadership, "Medical" = medical, "Surgery" = surgery, \
-	"Piloting" = pilot, "Police" = police, "Powerloaders" = powerloader, "Large Vehicles" = large_vehicle)
-
-
-/datum/skills/pfc
+	var/name = "Unassigned" //the name of the skills datum.
+	var/datum/mind/owner
+	var/datum/skillset/skillset
+
+	var/list/skill_list = ALL_SKILL_TYPES
+
+	var/list/skill_mods //contains the total skill modifier values. refer to __DEFINES/skills.dm for its usage in the macros.
+	var/list/skill_mod_ids //contains the applied skill modifier IDs, sorted by skill. used in macros too.
+
+	//used on /mob/proc/check_skills()
+	var/cached_skills_dat
+
+	//used to allow observers to bypass possible skill checks mainly found on examine().
+	var/omnighost = FALSE
```

Comment:
Perhaps a better name like "bypass"? Assuming this will allow admins to remove all restrictions easily